File: app/src/main/java/mono/hg/models/PinnedApp.java
package mono.hg.models;

import android.graphics.drawable.Drawable;
import android.view.View;

import androidx.annotation.NonNull;

import java.util.List;

import eu.davidea.flexibleadapter.FlexibleAdapter;
import eu.davidea.flexibleadapter.items.IFlexible;
import mono.hg.R;
import mono.hg.utils.AppUtils;

public class PinnedApp extends App {

    public PinnedApp(Drawable icon, @NonNull String packageName, long user) {
        super(null, packageName, false, user);
        setIcon(icon);
        setUserPackageName(AppUtils.appendUser(user, packageName));
    }

    public PinnedApp(String packageName, long user) {
        super(packageName, user);
        setUserPackageName(AppUtils.appendUser(user, packageName));
    }

    @Override public int getLayoutRes() {
        return R.layout.list_pinned_item;
    }

    @Override public PinnedApp.ViewHolder createViewHolder(View view, FlexibleAdapter<IFlexible> adapter) {
        return new ViewHolder(view, adapter);
    }

    @Override public void bindViewHolder(FlexibleAdapter<IFlexible> adapter,
            PinnedApp.ViewHolder holder, int position,
            List<Object> payloads) {
        holder.icon.setImageDrawable(getIcon());
    }
}

File: app/src/main/java/mono/hg/models/PinnedApp.kt
package mono.hg.models

import android.graphics.drawable.Drawable
import android.view.View
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.IFlexible
import mono.hg.R
import mono.hg.utils.AppUtils

/**
 * An extension of the [App] class that doesn't extracts app name.
 * This class is used in the favourites panel.
 */
class PinnedApp : App {
    constructor(icon: Drawable?, packageName: String, user: Long) : super(
        null,
        packageName,
        false,
        user
    ) {
        this.icon = icon
        userPackageName = AppUtils.appendUser(user, packageName)
    }

    constructor(packageName: String, user: Long) : super(packageName, user) {
        userPackageName = AppUtils.appendUser(user, packageName)
    }

    override fun getLayoutRes(): Int {
        return R.layout.list_pinned_item
    }

    override fun createViewHolder(
        view: View,
        adapter: FlexibleAdapter<IFlexible<*>?>?
    ): ViewHolder {
        return ViewHolder(view, adapter)
    }

    override fun bindViewHolder(
        adapter: FlexibleAdapter<IFlexible<*>?>?,
        holder: ViewHolder, position: Int,
        payloads: List<Any>
    ) {
        holder.icon.setImageDrawable(icon)
    }
}